docs/proposals/rdma/20240926- the end-to-end solution of RDMA devices.md

```diff
@@ -0,0 +1,172 @@
+
+
+# Supports the end-to-end solution of RDMA devices
+
+## Summary
+
+Currently, only CPU, memory, and GPU resource allocation is supported, and RDMA network device scheduling is not supported. The purpose of this proposal is to expand the end-side capabilities of RDMA devices, improve the scheduling and distribution capabilities of RDMA devices, and enhance the ability of container pass-through RDMA#PF or VF(SR-IOV) devices.
+
+## Motivation
+
+This involves RDMA device registration, resource reporting, updating node status, scheduling and allocation, annotating Pods, and modifying multi-NIC CNI plug-ins. The middle side does not rely on device-plugin, but continues the existing koordlet component. koordinator's scheduling component already supports RDMA and joint scheduling of RDMA and GPU, but the allocation result does not have the BDF information of PF, resulting in the failure to allocate PF to the container. In order to improve the end-to-end capability of RDMA equipment, it is necessary to put forward a set of solutions to realize the above functions and complete the whole process of RDMA equipment.
+
+### Goals
+
+1. Realize RDMA network device registration, including PF and VF. Add the PF/VF topology information of the RDMA device in DeviceCRD;
+2. Update the number of RDMA devices to the status field of the cluster node;
+3. Add the BDF address information for PF to the device assignment result on the Pod annotations;
+4. Modify the mutlus-cni plug-in source code, parse the DeviceId in the Pod annotation and inject it into the container;
+5. according to the label of the RDMA device, the scheduler is limited to complete the PF/VF scheduling and allocation in this range, such as: a GPU server has RDMA1 for GPUs, and RDMA2 for storage, Pod1 applies for GPU and RDMA device (PF), Pod2 applies for VF of RDMA2.
+
+### Non-Goals/Future Work
+
+1. None
+
+## User stories
+
+### Story 1
+
+As a user, I want to complete the joint scheduling of GPU and RDMA devices, and successfully mount the PF device inside the container and start it successfully.
+
+
+### Story 2
+
+As a user of a NIC device with the SR-IOV function, I want to request a certain number of VFs that meet the topology affinity, pass through the container, and let the container use VF to send and receive network traffic.
+
+## Proposal
+
+### Expand koordlet's ability to discover and register NIC devices
+
+Develop a device plug-in called RDMA that implements the DeviceCollector interface for collecting rdma device information.   
+
+```go
+package rdma
+
+const (
+   DeviceCollectorName = "RDMA"
+)
+
+type rdmaCollector struct {
+   enabled bool
+}
+
+func New(opt *framework.Options) framework.DeviceCollector {
+	return &rdmaCollector{
+		enabled: features.DefaultKoordletFeatureGate.Enabled(features.NetDevices),
```

Comment:
how do you plan to use this featuregate? if it should be named as features.NetDevices or just features.RDMADevice or something else?

do we have any experience here to name this considering the extensibility here? @ZiMengSheng